### DISCUSSION TITLE
More advanced options for print "by object"

### DISCUSSION THREAD

**Discussion by NotYeMK** [OrcaSlicer/OrcaSlicer#2174] [Ideas]
One of the issues that I have run into with Orca Slicer so far is that whenever I am printing by object and auto arrange everything, I don't have enough clearance for my Z Probe when printing objects right next to each other with the order and starting point that it goes in. 

I was hoping if some more features like being able to choose where the starting object is or be able to modify the order that it starts in, like being able to choose it from a drop-down menu, draw it out from the top down on the build plate, or even just creating the order based off a list. 

As a different option, maybe being able to set the clearance radiuses per side instead of all around. 

Basically just looking for a way to auto arrange for by object printing that can let me account for a z probe.

**Comment by CyberGaut** [comment]
clearance by side would be great
selectin the order or at least identifying the order, or at least explaining the rules used would be helpful.
right now you need to review the sliced plate and then play.

**Comment by CyberGaut** [comment]
Part of the print order and limits should also take into account the bed position and Bed type.
I get that a core XY needs to account for the height of the roof above the plate.
In a bed slinger if we print the objects from front to back (of the plate), the heights don't matter as all items will be in front of the print head and X axis bar.
**Thanks dev's for any and all considerations, what you do is magic! Much Apricated!**

**Comment by MikeHersee** [comment]
This is the same as the issue I have. My 3D printer has a very small distance from the print nozzle to the front of the assembly, a medium distance from the nozzle to the sides of the assembly, and a much bigger distance from the nozzle to the rear fo the assembly. Also, the beam supporting the head is behind the head. So if it were possible to specify the distance from the nozzle to each side, and the distance of the beam from the nozzle and the height of the beam above the nozzle, then by printing from the front backwards, it would be possible in most cases to fit a much larger number of objects on the print plate when printing by object. This would be important for me.
